### ISSUE TITLE
NVDA's support for Scintilla controls. is broken in alpha snapshots, if "line wrap" is enabled in the text editor

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#9424]
### Steps to reproduce:
1. Install a text editor like Notepad++.
2. In it, open a text file containing a few lines of text.
3. Enable the text editor's "line wrap" feature.
4. Start reading line by line with the up/down arrow keys.
5. Enable NVDA's mouse tracking and use the mouse to hover over and read the separate lines of text.

### Actual behavior:
With the up/down arrow keys, NVDA reads another line of text and not the one on which the cursor is positioned. This behavior affects only reading line by line - reading by word or character seams to be working as expected. The situation when reading with the mouse is similar - NVDA does not read the line which the mouse is pointing, but another one. And this occurs only if the "line wrap" feature of the text editor is enabled, so long lines of text are wrapped on multiple lines if necessary.

### Expected behavior:
NVDA should read the line on which the cursor is positioned or the one which the mouse is pointing, even if the "line wrap" feature of the text editor is enabled.

### System configuration
- Operating system: Windows 10 Pro version 1809 (build 17763.379), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-16976,7fc6c71f.
- Processor: Intel Core i5-4670 at 3.40GHz.
- RAM Memory: 8.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory + 4065MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC887 at Intel Lynx Point PCH - High Definition Audio Controller.

#### Does the issue still occur after restarting your PC?
Yes, it does.

#### Have you tried any other versions of NVDA?
No, because this started occurring only with the latest "alpha" snapshots of NVDA. I guess this is probably caused by #9233 (Fixing incorrect text reporting when hovering mouse over Scintilla controls), though I might be wrong.

**Comment by leonardder** [comment]
CC @DataTriny, this is probably caused by #9233. Could you please investigate as soon as possible what might causing this?

**Comment by Timtam** [comment]
reproducable by me, it also messes up the text shown on braille displays as well if thats of any help, tested with a Baum braille display addressed via Baum drivers.

**Comment by DataTriny** [comment]
Hi,
I can confirm that my PR introduced the bug mentioned here. I didn't see it while testing because I never enable word wrapping in Notepad++.

@leonardder the coordinates passed to `_getOffsetFromPoint` are correct, but the ones returned by `winUser.ScreenToClient` are not. I will investigate further to find whether it is a Scintilla bug or not.

**Comment by DataTriny** [comment]
Alright, fixed in #9427.

@leonardder feel free to tell me if I did something wrong. This part of the code were totally broken by the way!

**Comment by lpintes** [comment]
The line reporting is fixed, but say all is broken when long lines are present and word wrap is enabled.
It worked perfectly before.

**Comment by leonardder** [comment]
@lpintes: Please open a new issue for this, along with steps to reproduce.

**Comment by DataTriny** [comment]
Hello to all of you,
I opened #9521 to describe the problem reported by @lpintes, and I fixed it in #9522. @leonardder if you want to review it, it's a really really small change.
Thanks for reporting.
